### PR TITLE
CW 68 exchange automatic rate selector

### PR DESCRIPTION
**Pull request by OmarHatem28**

### CODE REVIEW

File: lib/src/screens/exchange/exchange_page.dart [hunk 1]
```diff
@@ -100,271 +100,197 @@ class ExchangePage extends BasePage {
...
                 ),
-                bottomSectionPadding:
-                    EdgeInsets.only(left: 24, right: 24, bottom: 24),
+                bottomSectionPadding: EdgeInsets.only(left: 24, right: 24, bottom: 24),
                 bottomSection: Column(children: <Widget>[
                   Padding(
                     padding: EdgeInsets.only(bottom: 15),
                     child: Observer(builder: (_) {
                       final description = exchangeViewModel.isFixedRateMode
+                          ? exchangeViewModel.isAvailableInSelected
```
Comment (OmarHatem28) [inline lib/src/screens/exchange/exchange_page.dart:282]: Sorry for the formatting, but the file was too un-organized to modify, anyway, here is the change that I added

File: lib/src/screens/exchange/exchange_page.dart [hunk 2]
```diff
@@ -399,8 +320,8 @@ class ExchangePage extends BasePage {
                           },
                           color: Theme.of(context).accentTextTheme.body2.color,
                           textColor: Colors.white,
-                          isLoading:
-                              exchangeViewModel.tradeState is TradeIsCreating)),
+                          isDisabled: exchangeViewModel.selectedProviders.isEmpty,
```
Comment (OmarHatem28) [inline lib/src/screens/exchange/exchange_page.dart:323]: here

File: lib/src/screens/exchange/exchange_page.dart (hunk 1 above)
Comment (OmarHatem28) [reply lib/src/screens/exchange/exchange_page.dart:282]: here is one

File: lib/view_model/exchange/exchange_view_model.dart [hunk 3]
```diff
@@ -77,10 +96,20 @@ abstract class ExchangeViewModelBase with Store {
   final Box<Trade> trades;
   final ExchangeTemplateStore _exchangeTemplateStore;
   final TradesStore tradesStore;
+  final SharedPreferences sharedPreferences;
 
   @observable
   ExchangeProvider provider;
 
+  /// Maps in dart are not sorted by default
```
Comment (OmarHatem28) [inline lib/view_model/exchange/exchange_view_model.dart:104]: I added documentation for most of the changes here, yet feel free to comment on anything that is unclear or needs more details

**Review by mkyq — APPROVED** [review]

**Pull request merged**